internal/pkg/deploy/cloudformation/app.go

```diff
@@ -104,36 +103,40 @@ func (cf CloudFormation) upgradeAppStackSet(config *stack.AppStackConfig) error
 	}
 }
 
-func (cf CloudFormation) upgradeAppStack(s *cloudformation.Stack) error {
-	for {
-		// Upgrade app stack.
+func newRenderAppStackInput(cfnStack *cloudformation.Stack) *executeAndRenderChangeSetInput {
```

Comment:
why do we prefer this over just embedding the code in `upgradeAppStack` 👀 ?

Reply:
haha this was more of a "copy the way it was done somewhere else" than me actually thinking about it. i removed it!!